pylint_nautobot/q_search_filter.py

```diff
@@ -0,0 +1,48 @@
+"""Check for use of SearchFilter on NautobotFilterSets instead of custom `q` search functions."""
+
+from astroid import Assign, AssignName, ClassDef
+from pylint.checkers import BaseChecker
+
+from .utils import is_version_compatible
+
+_META_CLASSES = {
+    "nautobot.extras.filters.NautobotFilterSet": ">1",
+    "nautobot.apps.filters.NautobotFilterSet": ">1",
+}
+
+
+class NautobotUseSearchFilterChecker(BaseChecker):
+    """Visit NautobotFilterSet subclasses and check for use of `q = SearchFilter`, instead of `q = django_filters.CharField`."""
+
+    version_specifier = ">=2,<3"
+
+    name = "nautobot-use-search-filter"
+    msgs = {
+        "C4272": (
```

Comment:
Do we have a constants.py or similar that can provide a central list of these codes to avoid reuse/conflicts?

Reply:
We do not, there is a PR for documenting these, but it hasn't been approved yet.

Reply:
I added a constants.py for these.